### PR TITLE
Decode xpriv if not present

### PR DESCRIPTION
**Pull request by ghostbuster91**
[trello](https://trello.com/c/pop2GFOQ/90-sakiewka-cryptosendcoins-powinien-opcjonalnie-przyjmowac-privkey-a-w-innym-przypadku-sprobowac-zdecodowac-klucz-klucz-prywatny-j)

### CODE REVIEW

File: src/lib/transaction.ts [hunk 1]
```diff
@@ -16,40 +16,66 @@ import {
...
+    throw <ErrorResponse>({ message: "Password or xprv has to be specified!", code: 400 })
+  }
+}
+
+const getUserXprvFromServer = async (wallet: GetWalletBackendResponse, userToken: string, password: string): Promise<string> => {
+  const keyId: string = wallet.keys.find(key => key.type === KeyType.USER)!.id
+  const key: GetKeyBackendResponse = await getKey(userToken, keyId, true)
+  const prvKey = key.prvKey
+  if (prvKey) {
+    return decrypt(password, prvKey)
```
Comment (adamw) [inline src/lib/transaction.ts]: if there's a decryption error, will the user get a nice error message?

File: src/lib/transaction.ts [hunk 2]
```diff
@@ -16,40 +16,66 @@ import {
...
+  } else {
+    throw <ErrorResponse>({ message: "There is no private key on server!", code: 400 })
```
Comment (adamw) [inline src/lib/transaction.ts]: should we return http-style errors here - what if this is used in the browser?

Comment (adamw) [reply src/lib/transaction.ts]: (as a library, not exposed as a server)

File: src/lib/transaction.ts [hunk 3]
```diff
@@ -29,14 +29,14 @@ const joinPath = (path: Path): string =>
   `${path.cosignerIndex}/${path.change}/${path.addressIndex}`
 
 export const send = async (
-  userToken: string, walletId: string, recipients: Recipient[], xprv?: string, password?: string
+  userToken: string, walletId: string, recipients: Recipient[], xprv?: string, passphrase?: string
```
Comment (adamw) [inline src/lib/transaction.ts]: why not `password`?

File: src/lib/transaction.ts (hunk 1 above)
Comment (ghostbuster91) [reply src/lib/transaction.ts]: I've just fixed it ;)

File: src/lib/transaction.ts (hunk 3 above)
Comment (ghostbuster91) [reply src/lib/transaction.ts]: Because I wanted to make a distinction between user's password and passphrase to encrypt wallet

**Pull request merged**